packages/vuetify/src/composables/icons.tsx

```diff
@@ -182,18 +182,23 @@ export const useIcon = (props: Ref<string | undefined> | { icon?: IconValue }) =
       }
     }
 
-    const hasSet = icon.includes(':')
-    const setName = hasSet ? icon.split(':')[0] : icons.defaultSet
-    const iconName = hasSet ? icon.split(':')[1] : icon
-    const set = icons.sets[setName ?? icons.defaultSet]
+    let iconSet: IconSet | undefined
+    let iconName: string | undefined
 
-    if (!set) {
-      throw new Error(`Could not find icon set "${setName}"`)
+    for (const iconSetName of Object.keys(icons.sets)) {
```

Comment:
I think I would prefer `Object.keys(icons.sets).find(...)` here instead of a for loop

Reply:
Like this?

```ts
const iconSetName = Object.keys(icons.sets).find(
  setName => typeof icon === 'string' && icon.startsWith(`${setName}:`)
)

const iconName = iconSetName ? icon.slice(iconSetName.length + 1) : icon
const iconSet = icons.sets[iconSetName ?? icons.defaultSet]

return {
  component: iconSet.component,
  icon: iconName,
}
```

Reply:
Yeah 👍 

Although is the typeof string necessary? at that point in the logic it should always be string I think?

Reply:
Within a callback context, TS cannot know whether the value/type of `icon` has changed, so its type is `IconValue | undefined`.

We could also do `(icon as string)`, but I try to avoid type assertions when possible.

Reply:
Should I push this change with the `typeof`, or do you have another preference?

Reply:
no it's fine as is